pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -0,0 +1,612 @@
+	}
+
+	// 2) Interact with registered Extenders.
+	strategies, err = CallExtenders(ph.Extenders(), pod, pl.fh.SnapshotSharedLister().NodeInfos(), strategies)
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+
+	// 3) Find the best strategy.
+	strategy := PickStrategy(strategies)
+	nominatedPodsToClear := strategy.NominatedPodsToClear()
+	if len(nominatedPodsToClear) == 0 {
+		nominatedPodsToClear = getLowerPriorityNominatedPods(ph, pod, strategy.NominatedNodeName())
+	}
+
+	// 4) Execute the best-fit PreemptionStrategy.
+
+	// Clear <nominatedPodsToClear> no matter <strategy.NominatedNode> is nil or not.
+	// It's because when a pod with nominated node name is eligible to preempt again, but preemption logic does not
+	// find any node for it, <strategy> returns the pod itself for removal of the 'NominatedNodeName' field.
+	for _, p := range nominatedPodsToClear {
+		if err := RemoveNominatedNodeName(cs, p); err != nil {
+			klog.Errorf("Cannot remove 'NominatedNodeName' field of pod: %v", err)
+			// We do not return as this error is not critical.
+		}
+	}
+
+	// Return if strategy.NominatedNodeName() is nil.
+	if len(strategy.NominatedNodeName()) == 0 {
+		return framework.NewStatus(framework.Unschedulable)
+	}
+
+	// Update the scheduling queue with the nominated pod information. Without
+	// this, there would be a race condition between the next scheduling cycle
+	// and the time the scheduler receives a Pod Update for the nominated pod.
+	// TODO(Huang-Wei): evaluate if it's needed after #90660.
+	ph.AddNominatedPod(pod, strategy.NominatedNodeName())
+
+	// Make a call to update nominated node name of the pod on the API server.
+	if err := SetNominatedNodeName(cs, pod, strategy.NominatedNodeName()); err != nil {
+		klog.Errorf("Error in %q plugin. Cannot set 'NominatedNodeName' on pod %v/%v: %v", Name, pod.Namespace, pod.Name, err)
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+
+	// Delete the victims on the API server.
+	for _, victim := range strategy.Victims().Pods {
+		if err := DeleteVictim(cs, victim); err != nil {
+			klog.Errorf("Error preempting pod %v/%v: %v", victim.Namespace, victim.Name, err)
+			return framework.NewStatus(framework.Error, err.Error())
+		}
+		// If the victim is a WaitingPod, send a reject message to the PermitPlugin
+		if waitingPod := pl.fh.GetWaitingPod(victim.UID); waitingPod != nil {
+			waitingPod.Reject("preempted")
+		}
+		// TODO: accommodate eventing.
+		// prof.Recorder.Eventf(victim, preemptor, v1.EventTypeNormal, "Preempted", "Preempting", "Preempted by %v/%v on node %v", preemptor.Namespace, preemptor.Name, nodeName)
+	}
+
+	return framework.NewStatus(framework.Schedulable)
+}
+
+// CalStrategies calculates a slice of Strategy. Executing any Strategy can make the given <pod> schedulable.
+func CalStrategies(ctx context.Context, state *framework.CycleState, pod *v1.Pod,
```

Comment:
Is `Cal` short for `Call` (like `CallExtenders` below) or `Calc` (like calculate)?

Reply:
It's short for "Calculate". Prefer "Calc"?

Reply:
ah, sure it's not a very big nit but I just wasn't sure if this was a typo